### PR TITLE
Update offset handling

### PR DESCRIPTION
**Pull request by fakob**
* Made Offset X and Offset Y
  * only affect where the object is drawn relative to the node
  * not have any effect downstream
* Updated the Combine objects node - BREAKING CHANGE
  * When plugging in graphics, an additional Position (TwoDVectorType) socket is added with which the objects can be aligned relative to each other
  * Added TwoDVectorType - for now it is just 2 number fields
  * It would be cool if the position could be changed the same way the offset is changed maybe with Alt+Drag, but I did not manage to make it work
  * Updated examples with the new node
* corrected inputPlugged outputPlugged notifications

https://github.com/fakob/plug-and-play/assets/4619772/e219bb3d-cdab-4f60-a609-d1d1b1281aed

### CODE REVIEW

File: src/nodes/draw/abstract.tsx [hunk 1]
```diff
@@ -189,10 +189,25 @@ export abstract class DRAW_Base extends PPNode {
     const drawingFunction = (
       container,
       executions,
-      offset = new PIXI.Point(),
+      position = new PIXI.Point(),
     ) => {
+      const lastNode = !this.getOutputSocketByName(outputPixiName).hasLink();
+      const drawnAsChildrenOfLastNode =
+        !lastNode &&
+        (executions === undefined || Object.keys(executions).length > 0);
```
[the remark this answers is not in the document]
Comment (fakob) [reply src/nodes/draw/abstract.tsx:197]: Let me investigate again. I ran into the issue where that actually happened.

File: src/nodes/draw/abstract.tsx [hunk 2]
```diff
@@ -189,10 +189,25 @@ export abstract class DRAW_Base extends PPNode {
...
+
+      const newOffset = new PIXI.Point(
```
[the remark this answers is not in the document]
Comment (fakob) [reply src/nodes/draw/abstract.tsx:199]: good idea

File: src/nodes/draw/abstract.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (magnificus) [reply src/nodes/draw/abstract.tsx:197]: hmm thats worrisome if so, do you know where that might come from?

File: src/nodes/draw/draw.tsx [hunk 3]
```diff
@@ -306,45 +308,97 @@ export class DRAW_Text extends DRAW_Base {
...
-  public socketShouldAutomaticallyAdapt(socket: Socket): boolean {
-    return true;
+
+  protected adaptInputs(): void {
+    const sockets = this.getAllNonDefaultInputSockets();
+    const offsetSockets = sockets.filter(
+      (socket) => socket.dataType instanceof TwoDVectorType,
+    );
+    sockets.forEach((socket) => {
+      const suffix = getSuffix(socket.name, outputPixiName);
```
[the remark this answers is not in the document]
Comment (fakob) [reply src/nodes/draw/draw.tsx:381]: It is a bit more complicated as I am also removing graphics sockets without links.

File: src/nodes/draw/abstract.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (fakob) [reply src/nodes/draw/abstract.tsx:197]: Right after removing the undefined check I had crash, but could not reproduce it anymore. I guess we will find out and then fix it :-)

File: src/nodes/draw/draw.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (fakob) [reply src/nodes/draw/draw.tsx:385]: I have now pushed an update where I did a small cleanup regarding this, but I tried your way and it did not work for me. I think it has something to do with the fact that I also have to remove the socket I just unplugged. Feel free to give it a try.

[the remark this answers is not in the document]
Comment (fakob) [reply src/nodes/draw/draw.tsx:385]: Sorry, no. Nothing I did made it more compact, elegant or more understandable :-|

[the remark this answers is not in the document]
Comment (magnificus) [reply src/nodes/draw/draw.tsx:385]: Ill take a look but feel free to merge

[the remark this answers is not in the document]
Comment (fakob) [reply src/nodes/draw/draw.tsx:385]: Will merge as I think it works fine, but thanks for looking at it!

**Pull request merged**